# Ruby 1.8.7 - Remove __inline from rb_reserved_word function
#
# This patch fixes a runtime symbol lookup issue on Android where the
# rb_reserved_word function was not being properly exported from the
# shared library due to the __inline compiler directive.
#
# Problem:
# - The rb_reserved_word function in lex.c was defined with __inline
# - This caused the function to not be exported to the dynamic symbol table
# - At runtime, dlopen would fail with "cannot locate symbol rb_reserved_word"
#
# Solution:
# - Remove the __inline directive to ensure the function is exported
# - This allows the symbol to be found at runtime
#
# Date: 2024-01-XX
# Tested on: Android NDK 25.2.9519653, arm64-v8a

diff --git a/lex.c b/lex.c
index a15309d..f704dc2 100644
--- a/lex.c
+++ b/lex.c
@@ -65,9 +65,6 @@ hash (str, len)
   return hval + asso_values[(unsigned char)str[len - 1]];
 }
 
-#ifdef __GNUC__
-__inline
-#endif
 struct kwtable *
 rb_reserved_word (str, len)
      register const char *str; 